docs/source/notes/amp_examples.rst

```diff
@@ -0,0 +1,186 @@
+.. _amp-examples:
+
+Automatic Mixed Precision Examples
+==================================
+
+.. currentmodule:: torch.cuda.amp
+
+.. contents:: :local:
+
+.. _gradient-scaling-examples:
+
+Gradient Scaling
+^^^^^^^^^^^^^^^^
+
+The code snippets below demonstrate recommended use of :class:`torch.cuda.amp.AmpScaler`.
+The :class:`AmpScaler` instance ``scaler`` performs dynamic gradient scaling
+(maintains the scale, helps create scaled gradients to prevent
+gradient underflow, and carries out scaling-safe steps).
+
+Typical use (1 loss, 1 optimizer)
+---------------------------------
+
+::
+
+    # Create an AmpScaler instance.
+    scaler = AmpScaler()
+    ...
+    for input, target in data:
+        optimizer.zero_grad()
+        output = model(input)
+        loss = loss_fn(output, target)
+
+        # Scale the loss, and call backward() on the scaled loss to create scaled gradients.
+        scaler.scale(loss).backward()
+
+        # Carry out a scaling-safe step.  scaler.step() unscales the optimizer's gradients
+        # and skips optimizer.step() if the gradients contain infs or NaNs.
+        scaler.step(optimizer)
+
+        # Update the scale for next iteration.
+        scaler.update()
+
+
+Scale-Aware Operations
+----------------------
+
+The gradients resulting from ``scaler.scale(loss).backward()`` are scaled.  ``scaler.step(optimizer)``
```

Comment:
Revise this paragraph to focus exclusively on the code delta and how scaling can be disabled. Key points:

- There are 4 lines of code changed. 
- It can be disabled/enabled with a flag. 
- update() should be called after each optimizer is stepped (more on that below?)